File: matchmaker/server/src/main/java/ru/ifmo/se/dating/matchmaker/api/HttpStatisticsApi.java
package ru.ifmo.se.dating.matchmaker.api;

import java.util.Set;

import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import ru.ifmo.se.dating.matchmaker.model.generated.StatisticsAttitudesGet200ResponseInner;
import ru.ifmo.se.dating.matchmaker.api.generated.StatisticsApiDelegate;

@Service
class HttpStatisticsApi implements StatisticsApiDelegate {

    @Override
    public ResponseEntity<Set<StatisticsAttitudesGet200ResponseInner>>
    statisticsAttitudesGet() {
        return ResponseEntityStub.create();
    }
}

File: matchmaker/server/src/main/kotlin/ru/ifmo/se/dating/matchmaker/api/HttpStatisticsApi.kt
package ru.ifmo.se.dating.matchmaker.api

import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Service
import ru.ifmo.se.dating.matchmaker.api.generated.StatisticsApiDelegate
import ru.ifmo.se.dating.matchmaker.model.generated.StatisticsAttitudesGet200ResponseInner

typealias StatisticsAttitudesResponse =
    ResponseEntity<List<StatisticsAttitudesGet200ResponseInner>>

@Service
internal class HttpStatisticsApi : StatisticsApiDelegate {
    override fun statisticsAttitudesGet(): StatisticsAttitudesResponse =
        ResponseEntityStub.create()
}
